src/managers/GuildScheduledEventManager.js

```diff
@@ -0,0 +1,261 @@
+
+    return data.reduce(
+      (coll, rawGuildScheduledEventData) =>
+        coll.set(
+          rawGuildScheduledEventData.id,
+          this.guild.scheduledEvents._add(rawGuildScheduledEventData, options.cache),
+        ),
+      new Collection(),
+    );
+  }
+
+  /**
+   * Options used to edit a guild scheduled event.
+   * @typedef {Object} GuildScheduledEventEditOptions
+   * @property {string} [name] The name of the guild scheduled event
+   * @property {DateResolvable} [scheduledStartTime] The time to schedule the event at
+   * @property {?DateResolvable} [scheduledEndTime] The time to end the event at
+   * @property {PrivacyLevel|number} [privacyLevel] The privacy level of the guild scheduled event
+   * @property {GuildScheduledEventEntityType|number} [entityType] The scheduled entity type of the event
+   * @property {string} [description] The description of the guild scheduled event
+   * @property {?GuildVoiceChannelResolvable} [channel] The channel of the guild scheduled event
+   * @property {GuildScheduledEventStatus|number} [status] The status of the guild scheduled event
+   * @property {GuildScheduledEventEntityMetadataOptions} [entityMetadata] The entity metadata of the
+   * guild scheduled event
+   * <warn>This is required if `entityType` is 'EXTERNAL'</warn>
+   */
+
+  /**
+   * Edits a guild scheduled event.
+   * @param {GuildScheduledEventResolvable} guildScheduledEvent The guild scheduled event to edit
+   * @param {GuildScheduledEventEditOptions} options Options to edit the guild scheduled event
+   * @returns {Promise<GuildScheduledEvent>}
+   */
+  async edit(guildScheduledEvent, options) {
+    const guildScheduledEventId = this.resolveId(guildScheduledEvent);
+    if (!guildScheduledEventId) throw new Error('GUILD_SCHEDULED_EVENT_RESOLVE');
+
+    if (typeof options !== 'object') throw new TypeError('INVALID_TYPE', 'options', 'object', true);
+    let {
+      privacyLevel,
+      entityType,
+      channel,
+      status,
+      name,
+      scheduledStartTime,
+      description,
+      scheduledEndTime,
+      entityMetadata,
+    } = options;
+
+    if (typeof privacyLevel === 'string') privacyLevel = PrivacyLevels[privacyLevel];
+    if (typeof entityType === 'string') entityType = GuildScheduledEventEntityTypes[entityType];
+    if (typeof status === 'string') status = GuildScheduledEventStatuses[status];
+
+    const data = await this.client.api.guilds(this.guild.id, 'scheduled-events', guildScheduledEventId).patch({
+      data: {
+        channel_id: typeof channel === 'undefined' ? channel : this.guild.channels.resolveId(channel),
+        name,
+        privacy_level: privacyLevel,
+        scheduled_start_time: scheduledStartTime ? new Date(scheduledStartTime).toISOString() : undefined,
+        scheduled_end_time: scheduledEndTime ? new Date(scheduledEndTime).toISOString() : scheduledEndTime,
+        description,
+        entity_type: entityType,
+        status,
+        entity_metadata: {
```

Comment:
Entity metadata can only be sent if the event is external, otherwise it causes an error

Reply:
CC @iShibi, if that's so, we should probably only append entity_metadata if the user provides it

Reply:
Oh if you want the error:
```
DiscordAPIError: Invalid Form Body
entity_metadata: This type of event should not have entity metadata.
```

Reply:
This seems to be happening with `VOICE` only, for `STAGE_INSTANCE` it just ignores it, which is how it should be I guess.

Reply:
Yeah it did happen on a VOICE event for me, should still be fixed regardless

Reply:
This issue has been fixed

Reply:
Fixed how?

Reply:
oopsie, added the fix only to `create`, should be there in the next commit for `edit` too